src/components/Dashboard.vue

```diff
@@ -128,290 +129,300 @@
-      } else {
-        console.log(`session terminated`)
-        window.location.hash = ''
-        window.location.reload()
-      }
+
+  import UserService from 'src/services/UserService'
+  import SessionService from 'src/services/SessionService'
+  import ListSessions from 'src/components/ListSessions'
+
+  import router from '../router'
+
+  export default {
+    components: {
+      ListSessions
     },
-    getHelp () {
-      this.popUpStyle = {
-        display: 'flex'
+    data() {
+      const user = UserService.getUser() || {}
+      SessionService.getCurrentSession(this, user)
+
+      const subtopics = {
+        math: ['Algebra', 'Geometry', 'Trigonometry', 'Precalculus', 'Calculus'],
+        esl: ['General Help'],
+        college: ['Planning', 'Applications', 'Essays']
+
+        // Temporarily changing to single word labels
+        // 'college': ['College Planning', 'Application Help','Essay Editing']
+
+        // Temporarily removing science and standardized testing
+        // 'science': ['Biology','Chemistry'],
+        // 'standardizedtest': ['SAT']
       }
-      this.coverStyle = {
-        background: 'rgba(0,0,0,0.10)'
+      return {
+        user,
+        name: user.firstname || 'student',
+        popUpStyle: {},
+        showHelpPopUp: false,
+        pickedTopic: '',
+        pickedSubtopic: '',
+        subtopics,
+        coverStyle: {}
       }
-      this.showHelpPopUp = true
-    },
-    capitalize (string) {
-      return string.charAt(0).toUpperCase() + string.slice(1)
     },
-    getHelpCancel () {
-      this.popUpStyle = {}
-      this.coverStyle = {}
-      this.showHelpPopUp = false
+    watch: {
+      // Watch the help topic for changes, and reset the subtopic when it does.
+      pickedTopic: function () {
+        this.pickedSubtopic = ''
+      }
     },
-    getHelpNext () {
-      let topic = this.pickedTopic
-      let subTopic = this.pickedSubtopic
-      // Temp change all to math
-      // topic = 'math';
-      topic = topic.toLowerCase()
-      subTopic = subTopic.toLowerCase()
-      if (subTopic === 'general help') {
-        subTopic = topic
+    methods: {
+      hasActiveSession() {
+        return localStorage.getItem('currentSessionPath')
+      },
+      rejoinHelpSession() {
+        const path = localStorage.getItem('currentSessionPath')
+        if (path) {
+          window.location.hash = `#${path}`
+          window.location.reload()
+          console.log(`rejoining session: ${path}`)
+        } else {
+          console.log(`session terminated`)
+          const user = UserService.getUser()
+          if (!user.isVolunteer) {
+            console.log('updating URL')
+            const url = '/feedback/null/student/null/null'
+            router.replace(url)
+          }
+          if (user.isVolunteer) {
+            console.log('updating URL')
+            const url = '/feedback/null/volunteer/null/null'
+            router.replace(url)
+          }
```

Comment:
Also, it's generally a good idea to try to avoid repeating logic when possible. In this case, you could put a single `console.log` statement before the first `if` statement, and you won't have to repeat it.

Reply:
Thanks for reviewing @dmnisson 

This code is outdated now as the file structures have changes. I will work on fixing the issues.

Also I believe I used two back to back "if" blocks as a way to make sure the **conditions are checked for both student and volunteer sessions**. Using if, else blocks would tackle just either student or volunteer sessions.